src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,7 @@
 
 import java.util.List;
 
-public interface MachineProducer {
-    /**
-     * In your implementations - create 2-3 models, add them to the list and return
-     * @return - the list of models
-     */
-    List<Object> get();
+public interface MachineProducer<T> {
```

Comment:
The MachineProducer interface should be parameterized with a type that extends Machine to prevent non-machine types from being used. For example: 'MachineProducer<T extends Machine>'.

Reply:
extends from Machine